api/src/main/java/org/openmrs/api/TagService.java

```diff
@@ -0,0 +1,156 @@
+ */
+
+public interface TagService extends OpenmrsService {
+	
+	/**
+	 * Returns a tag by uuid. It can be called by any authenticated user. It is fetched in read only
+	 * transaction.
+	 * 
+	 * @param uuid
+	 * @return tag or null
+	 */
+	@Authorized(TagConstants.GET_TAGS)
+	Tag getTagByUuid(String uuid);
+	
+	/**
+	 * Saves a tag. Sets the owner to superuser, if it is not set. It can be called by users with
+	 * this module's privilege. It is executed in a transaction. Tags can only be added, not edited.
+	 * 
+	 * @param tag
+	 * @return Tag that has been saved
+	 */
+	@Authorized(TagConstants.MANAGE_TAGS)
+	Tag saveTag(Tag tag);
+	
+	/**
+	 * Completely delete the tag from the database.
+	 * 
+	 * @param tag The Tag to remove from the system
+	 */
+	@Authorized(TagConstants.MANAGE_TAGS)
+	void deleteTag(Tag tag);
+	
+	/**
+	 * Removes a tag(if exists) from an OpenmrsObject.
+	 * 
+	 * @param openmrsObject the openmrsObject from which the tags is to be removed
+	 * @param tag the textual tag to be removed
+	 */
+	@Authorized(TagConstants.MANAGE_TAGS)
+	void removeTag(OpenmrsObject openmrsObject, String tag);
+	
+	/**
+	 * Returns a tag by id.
+	 * 
+	 * @param tagId the id of the Tag Object to be retrieved
+	 * @return Tag with given internal identifier
+	 */
+	@Authorized(TagConstants.GET_TAGS)
+	Tag getTag(Integer tagId);
+	
+	/**
+	 * Returns a list of Tag objects having the given tag, or having tags containing the substring.
+	 * Partial Search supported
+	 * 
+	 * @param tag the tag (String) which the
+	 * @return List of Tag Objects
+	 */
+	@Authorized(TagConstants.GET_TAGS)
+	List<Tag> getTags(String tag);
+	
+	/**
+	 * Returns a list of All Tags.
+	 * 
+	 * @return list Of Tags (String)
+	 */
+	@Authorized(TagConstants.GET_TAGS)
+	List<String> getAllTags();
+	
+	/**
+	 * Retrieves an Object with given object uuid and object type from the Openmrs database.
+	 * 
+	 * @param objectUuid the UUID of the object
+	 * @param objectType the java class of the object (eg. Concept.class)
+	 * @return
+	 */
+	<T extends OpenmrsObject> T getObject(Class<T> objectType, String objectUuid);
+	
+	/**
+	 * It adds a tag to the openmrs object, if such a tag does not already exist on the object. A
+	 * warning will be logged if the object does not exist in the database
+	 * 
+	 * @param openmrsObject the object to which the tag has to be added
+	 * @param tag the tag (string) which is to be added
+	 */
+	@Authorized(TagConstants.GET_TAGS)
+	Tag addTag(OpenmrsObject openmrsObject, String tag);
+	
+	/**
+	 * It returns a list of tags added to the openmrs object
+	 * 
+	 * @param openmrsObject the object whose tags needs to obtained
+	 * @return a List of Tag Objects having the same object type and uuid as the Openmrs Object
+	 *         parameter
+	 */
+	@Authorized(TagConstants.GET_TAGS)
+	List<Tag> getTags(OpenmrsObject openmrsObject);
+	
+	/**
+	 * returns a List of Tag Objects which have an Object Type that belongs to the objectTypes
```

Comment:
Please update all your javadocs to start with uppercase letter, it also feels strange to start with returns